### ISSUE TITLE
VAE expects inputs with batch dim divisible by batch size

### ISSUE THREAD

**Issue by miketynes** [miketynes/AttentiveWorldModels#6]
Due to some unfortunate quirks of our keras implementation, the VAE expects inputs in batches divisible by batch size, even at test time. This is clearly unsustainable, and has already lead to messy workarounds. I propose we rewrite in the style of the tensorflow docs [found here](https://www.tensorflow.org/tutorials/generative/cvae)
